text/0249-v2-experiments.md

```diff
@@ -374,6 +380,257 @@ Disadvantages:
+`Bucket` type:
+
+**Advantages of adding the experimental addition as a prefix:**
+
+1. Might be more clear that the API is experimental. All experimental APIs are
+   grouped together in autocomplete.
+2. When the final API is added, the autocomplete experience will be less
+   confusing. `grantWrite` will be visually separated than `preV1GrantWrite` and
+   `preV2GrantWrite`.
+
+**Advantages of adding the experimental addition as a suffix:**
+
+1. Discoverability. When using the `Bucket` type in code, a user looking for a
+   way to "grant" permission on the `Bucket`, is likely to type "grant" in the
+   IDE and look in the suggestion displayed by the autocomplete. A method that
+   starts with `grant` will be listed before a method that ends with it.
+   However, most IDEs will list methods that contains "grant" in any part will
+   be listed by autocomplete when a user types "grant".
+
+#### Alternatives:
+
+Using `v1` for visibility:
+
+- RcV1: for release candidate.
+- AlphaV1
+
+**How will it look in all supported languages:**
+
+> _to be completed once a naming scheme is decided_
+
+| scheme | TS  | Java | Python | .Net | Go  |
+| ------ | --- | ---- | ------ | ---- | --- |
+| PreV1  |     |      |        |      |     |
+
+### Discussion
+
+Advantages:
+
+- Experimental APIs can be used without declaring a fixed version on
+  `aws-cdk-lib`.
+- We will be able to push critical updates to previous experimental versions of
+  an API. For example, if we discover that `grantWrite` grants overly permissive
+  permissions, and the same occur in all of its experimental (deprecated)
+  predecessors, we will be able to push the fix to them as well. This will allow
+  us to get the fix to more customers in case of a critical fix.
+- All APIs will be in `aws-cdk-lib`, preventing dependencies hell.
+- Libraries will be able to use experimental APIs without locking their
+  consumers to a specific version of `aws-cdk-lib`.
+- Same solution across all languages.
+
+Disadvantages:
+
+- User does not have a clear motivation to upgrade to a newer version.
+- Low adoption. Users might be worried to use an API with an experimental name,
+  assuming that experimental means "will break".
+- Cluttering the code base. Although most IDEs will mark deprecated properties
+  with a ~~strikethrough~~, they will still be listed by autocomplete.
+- Will force some pretty long and ugly names on our APIs. Many experimental APIs
+  will result in a less aesthetic user code.
+- Graduating a module will require a lot of code changes from our users to
+  remove the prefix/suffix.
+- A lot of deprecated code in aws-cdk-lib, possibly blowing up `aws-cdk-lib`.
+  This might not be a real concern as we can reuse a lot the code between
+  different version of an API.
+
+**How can we encourage users to upgrade to a newer version?**
+
+When a new minor version introduces a breaking change to an experimental API,
```

Comment:
Search & replace "experimental" with "preview" in this section, so we can "feel" if this semantics works:

```suggestion
When a new minor version introduces a breaking change to an preview API,
```